### PR TITLE
Adding basic auth option for gateway

### PR DESCRIPTION
**Pull request by KrustyHack**
Signed-off-by: Nicolas Hug <nicolas.hug@proximis.com>

## Description

I've added the option to activate basic auth for gateway as described here : https://docs.openfaas.com/reference/authentication/

## Motivation and Context

It wasn't possible to deploy OpenFaas chart with basic auth activated on the gateway because option wasn't implemented.

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Helm v2.9.1 and Kubernetes v1.10.4-gke.0.
I've tested deployment in our GKE cluster, with and without basic auth.
I've tested functions deployment with and without basic auth through faas-cli and browser.
I've tested functions with and without basic auth by calling them with faas-cli and with browser.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: chart/openfaas/README.md [hunk 1]
```diff
@@ -51,6 +51,18 @@ $ helm upgrade --install openfaas openfaas/ \
    --set functionNamespace=openfaas-fn
 ```
 
+## Activate basic auth for gateway
+
+Create a Kubernetes secret named `basic-auth` in `openfaas` namespace :
+
+```
+k create secret generic basic-auth --from-literal=user=test --from-literal=password=test
```
Comment (alexellis) [inline chart/openfaas/README.md]:
```
-bash: k: command not found
```

I think `k` is not a standard alias for `kubectl`

File: chart/openfaas/templates/gateway-dep.yaml [hunk 2]
```diff
@@ -28,8 +28,17 @@ spec:
...
+        volumeMounts:
+        - name: basic-auth
+          mountPath: "/secrets"
+          readOnly: true
+        {{- end }}
         env:
+        - name: basic_auth
+          value: "{{ .Values.gateway.basicAuth }}"
+        - name: secret_mount_path
+          value: /secrets
```
Comment (alexellis) [inline chart/openfaas/templates/gateway-dep.yaml]: Can you make this `/var/run/openfaas/secrets` to match what we use for functions?

File: chart/openfaas/README.md (hunk 1 above)
Comment (KrustyHack) [reply chart/openfaas/README.md]: My bad it's my own alias. I will change it.

File: chart/openfaas/templates/gateway-dep.yaml (hunk 2 above)
Comment (KrustyHack) [reply chart/openfaas/templates/gateway-dep.yaml]: Yes

Comment (KrustyHack) [reply chart/openfaas/templates/gateway-dep.yaml]: Fixed in second commit. :)

File: chart/openfaas/README.md (hunk 1 above)
Comment (KrustyHack) [reply chart/openfaas/README.md]: Fixed in second commit. :)

**Comment by alexellis** [comment]
Let's get rid of that unsigned merge commit.  

```
git reset HEAD~2
git add .
git commit -s
git push origin master --force
```

Alternatively close this one and raise a new PR with a specifically named branch `git checkout -b basic-auth-helm`

**Comment by alexellis** [comment]
Hey @KrustyHack I appreciate you working on this. I think we have this merged now via other work done by @stefanprodan. I'll close, but feel free to contribute anything that was missing in Stefan's commits.

Alex

**Pull request closed without merging**